nether/portal.lua

```diff
@@ -79,11 +100,13 @@ end
 
 -- used for obsidian portal
 local function obsidian_teleport(player, pname, target)
-	minetest.chat_send_player(pname, "For any reason you arrived here. Type " ..
-		"/nether_help to find out things like craft recipes.")
-	players_in_nether[pname] = true
-	save_nether_players()
-	update_background(player, true)
+	if nether.trap_players then
```

Comment:
If I understand my messy code correctly, this is the wrong way to fix #16. 
Before the changes, if damage is enabled, the obsidian portal kills players even if `nether.trap_players` is disabled. If `nether.trap_players` is disabled, the `minetest.register_on_respawnplayer` is not executed, so the died player remains in the overworld.

So instead of changing `obsidian_teleport`, I'd change `obsi_teleport_player` so that it executes `obsidian_teleport` without target position only if both damage and `nether.trap_players` are enabled and otherwise executes it with target position:
```
	local has_teleported
	if damage_enabled and nether.trap_players then
		obsidian_teleport(player, pname)
		has_teleported = true
	elseif not mclike_portal then
		local target = vector.round(get_player_died_target(player))
		if generated_or_generate(target) then
			obsidian_teleport(player, pname, target)
			has_teleported = true
		end
	end
```
I have not tested if my suggestion works.

Reply:
Okay. Found it. I tried to stay out of the messy stuff near the bottom. :D

If I understand this correctly, solving #16 this way will no longer kill the player if they try to use an Obsidian portal with `nether.trap_players = false`. Is that what you intended? It makes sense to me, but my previous impression is that you wanted to kill any player attempting to use an Obsidian portal, no matter what (assuming damage enabled).

My next pull request will have your suggested solution. Let me know if that is not what you intended, and I will switch it back.

Reply:
Yes, this is what I intended, thanks. 
If damage is enabled and `nether.trap_players` is disabled and the Obsidian portal kills the player, the player cannot use it to get to the nether, so I think in this case the player should be teleported instead of killed.